### PR TITLE
Crash when filtering numeric values as empty strings. #1194

### LINKED ISSUE #1194: Crash in PostgreSQL when filtering with empty value (v6.0.5)
With querystring `?last_modified=`

Error 
```
(psycopg2.DataError) invalid input syntax for type numeric: ""
```
From this `get_all()` query
```
AND (data->>%(filters_field_0_0)s)::numeric = %(filters_value_0)s
```
that becomes:
```
...
AND (data->>'1492639357')::numeric = ''
...
```

### PR DESCRIPTION
**Pull request by Natim**
Fixes #1194 

- [X] Add tests.
- [x] Add a changelog entry.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -70,6 +70,7 @@ Protocol is now at version **1.16**. See `API changelog`_.
   bucket (#1137)
 - Removed Structlog binding and bottlenecks (fixes #603)
 - Fixed Swagger output with subpath and regex in pyramid routes (fixes #1180)
+- Fixed Postgresql errors when trying to compare numeric value with an empty string. (fixes #1194)
```
Comment (leplatrem) [inline CHANGELOG.rst]: → When specified empty values in querystring filters

File: kinto/core/storage/postgresql/__init__.py [hunk 2]
```diff
@@ -703,7 +703,8 @@ def _format_conditions(self, filters, id_field, modified_field,
                 sql_field = "coalesce({}, '')".format(column_name)
                 # Cast when comparing to number (eg. '4' < '12')
                 try:
-                    if is_numeric(value) or all([is_numeric(v) for v in value]):
+                    # all([]) is True while all(None) raises a TypeError.
+                    if is_numeric(value) or all([is_numeric(v) for v in value] or None):
```
Comment (leplatrem) [inline kinto/core/storage/postgresql/__init__.py]: Why not 👍  :) But we could also write `if value and (is_numeric(value) or all([...]))` no?

Comment (Natim) [inline kinto/core/storage/postgresql/__init__.py]: Yes sure.

File: CHANGELOG.rst (hunk 1 above)
Comment (Natim) [inline CHANGELOG.rst]: For me the fix is at the storage level not the request level, but if it feels more natural, I can change the description.

**Pull request merged**